File: android/lib/model/kategori.dart
import 'package:yunikah/model/asset.dart';

class StatusKategori {
  static const Rias = 'rias';
  static const Photo = 'photo';
  static const Gedung = 'gedung';
  static const AlatPesta = 'alat';
  static const Undangan = 'undangan';

  int id;
  String keterangan;

  StatusKategori({
    this.id,
    this.keterangan,
  });

  factory StatusKategori.parseFromJson(Map<String, dynamic> data) => StatusKategori(
    id: data['id'],
    keterangan: data['keterangan'],
  );
}

class Kategori {
  int id;
  String name;
  int harga;

  Asset image;

  StatusKategori statusKategori;

  Kategori({
    this.id,
    this.name,
    this.harga,
    this.image,
    this.statusKategori
  });

  factory Kategori.parseFromJson(Map<String, dynamic> data) => Kategori(
    id: data['id'],
    name: data['name'],
    harga: data['harga'],
    image: Asset.parseFromJson(data['image']),
    // statusKategori: StatusKategori.parseFromJson(data['status'])
  );
}

List<Kategori> kategoriFromJson(List<dynamic> data) => data.map((e) => Kategori.parseFromJson(e)).toList();

File: AndroidStudio/app/src/main/java/com/kosim/yunikah/model/Kategori.kt
package com.kosim.yunikah.model

import com.google.gson.annotations.SerializedName

data class StatusKategori (
    @SerializedName("id")
    var id: Int?,
    @SerializedName("keterangan")
    var keterangan: String?
) : Response()

data class Kategori(
    @SerializedName("id")
    var id : Int?,
    @SerializedName("name")
    var name : String?,
    @SerializedName("harga")
    var harga: Int?,
    @SerializedName("image")
    var image: Asset?

) : Response()